price-estimator/src/filter.rs

```diff
@@ -78,15 +78,17 @@ async fn estimate_buy_amount<T>(
 }
 
 async fn get_markets<T>(
-    _token_pair: TokenPair,
+    token_pair: TokenPair,
     _query: QueryParameters,
-    _orderbook: Arc<Orderbook<T>>,
+    orderbook: Arc<Orderbook<T>>,
 ) -> Result<impl warp::Reply, Infallible> {
-    // TODO: implement
-    Ok(warp::reply::json(&MarketsResult {
-        asks: Vec::new(),
-        bids: Vec::new(),
-    }))
+    let transitive_orderbook = orderbook.get_pricegraph().await.transitive_orderbook(
+        token_pair.sell_token_id,
+        token_pair.buy_token_id,
```

Comment:
To get the same results as the nodejs price estimator I had to put sell_token as the first argument here. Not sure if this is expected. This is clear when trying a token pair that only either has asks or bids but not the other like `1-10`.

Reply:
Hmm, this may be an issue with the direction of ask and bid orders... They *might* be inverted in *pricegraph*. @fleupold , do you know if "ask" orders are "buying base token and selling quote token" or "buying quote token and selling base token"? I'm pretty sure "pricegraph" uses the first definition (at least, that is the intention).

Reply:
If you have `/markets/x-y/estimated-buy-amount/1` return something (not `None`), that means you can sell y for x, so should `/markets/x-y/` be guaranteed to have at least one bid or at least one ask?

Reply:
> at least one bid or at least one ask?

That is the question!

Reply:
According to wikipedia:
> Bids (buyers) on the left, asks (sellers)

And since the price is in quote token, to me this implies bids are buying base token and selling quote token, and asks are buying quote token and selling base token. So the `pricegraph` has it backwards.

I will wait on @fleupold to confirm before making `pricegraph` changes.

Eitherway, I think the base token should be the "buy" token, and the quote token should be the "sell" token in the call to `Pricegraph::transitive_orderbook` (as that was the expectation), and that the direction should be fixed internally in the `pricegraph` crate.

Reply:
If you are looking at an ETH/USD orderbook you can buy ETH from the asks for USD. Thus the asks buying USD (quote token) and selling ETH (base token). So @nlordell, your statement above is correct. This is another thing that is super ambiguous and I always have to look at existing orderbooks for this.

Reply:
I am going to merge this then as it is correct right now. Then in separate PR we will switch buy/sell here and in the pricegraph so that is stays correct but has the expected parameter naming.